### DISCUSSION TITLE
Confusing module import/export behavior

### DISCUSSION THREAD

**Discussion by Jack-Above** [webpack/webpack#16402] [Q&A] [answered]
Hello all,

I am building a little SPA with React + Material UI. I have installed the default material packages but things have gone awry when importing components.

### What is the current behavior?
Running the webpack dev server with `npm start` -> `webpack serve --hot --open` produces the following warning:
```
WARNING in ./node_modules/@mui/material/esm/SvgIcon/SvgIcon.js 24:31-53
export 'getSvgIconUtilityClass' (imported as 'getSvgIconUtilityClass') was not found in './svgIconClasses' (possible exports: default)
```

### What is the expected behavior?
I can browse the subject files and (given they are only a few lines long) can clearly see a named export (`getSvgIconUtilityClass`) alongside a default export. However, I am interpreting the above error to explicitly contradict this.

./node_modules/@mui/material/esm/SvgIcon/svgIconClasses.js:
```js
import { generateUtilityClass, generateUtilityClasses } from '@mui/base';
export function getSvgIconUtilityClass(slot) {
  return generateUtilityClass('MuiSvgIcon', slot);
}
const svgIconClasses = generateUtilityClasses('MuiSvgIcon', ['root', 'colorPrimary', 'colorSecondary', 'colorAction', 'colorError', 'colorDisabled', 'fontSizeInherit', 'fontSizeSmall', 'fontSizeMedium', 'fontSizeLarge']);
export default svgIconClasses;
```

I have actually used the same mui version in a create-react-app before with no issue.. I suspect there is some configuration that was hidden by CRA that I am unable to find (because the ejected configs are overwhelming to a noob like me).

Any suggestions are appreciated. Thanks in advance!

### Other relevant information:
- webpack version: ^5.74.0
- webpack-cli: ^4.10.0
- webpack-dev-server: ^4.11.1
- @mui/material: ^5.10.8
- Node.js version: v16.16.0
- Operating System: Ubuntu 20.04 LTS
- Additional tools: (I'm not sure what else is relevant; will happily add)

**Comment by alexander-akait** [comment, marked as answer]
Do you have babel?

**Reply by Jack-Above** [reply]
Yes, using babel. Versions are:
```json
    "@babel/core": "^7.19.3",
    "@babel/preset-env": "^7.19.3",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
```

and .babelrc:
```json
{
  "presets": [
    "@babel/preset-env",
    [
      "@babel/preset-react",
      {
        "runtime": "automatic"
      }
    ],
    "@babel/preset-typescript"
  ]
}

**Reply by alexander-akait** [reply]
Can you provide tsconfig? Looks like you convert ESM code to common js, so you have `default`

**Reply by Jack-Above** [reply]
Sure, tsconfig.json is:
```json
{
  "compilerOptions": {
    "target": "ES5" ,
    "module": "ESNext" ,
    "moduleResolution": "node" ,
    "lib": [
      "DOM",
      "ESNext"
    ] ,
    "jsx": "react-jsx" ,
    "noEmit": true ,
    "isolatedModules": true ,
    "esModuleInterop": true ,
    "strict": true ,
    "skipLibCheck": true ,
    "forceConsistentCasingInFileNames": true ,
    "resolveJsonModule": true
  },
  "include": [
    "src/**/*"
  ]
}
```
Thanks for taking your time to look at this

**Reply by alexander-akait** [reply]
Can you provide webpack configuration? WIll be great if you can create minimum reproducible test repo to avoid such questions and I will show you the problem and how to fix it faster

**Reply by Jack-Above** [reply]
Good idea on the test repo, made one [here](https://github.com/Jack-Above/broken_mui_example). Just as a reminder I'm using node 16.16

**Reply by alexander-akait** [reply]
Here some notes:
1. You don't need `webpack.HotModuleReplacementPlugin`, it will be enabled by default when you use `hot: true`
2. Please fir your RegExp to `/\.(png|svg|jpg|gif)$/`, otherwise `/\.png|svg|jpg|gif$/` capture `SvgIcon.js` file (do the same for all regexps, i.e. parens)

**Reply by Jack-Above** [reply]
Wow great catch... the regex fixed it. I'm not sure I would have ever landed on that solution. Thanks a lot for your help!

For anyone else reading this, I'll leave that repo up. The changes he is referring to are in [webpack.config.js](https://github.com/Jack-Above/broken_mui_example/blob/70b75664490725d9a79aa5b9b814d0eeb8c84970/webpack.config.js) (lines 23 & 47).
